api/commands/run.ts

```diff
@@ -99,7 +100,7 @@ export class Runner {
     this.eEmitter.setMaxListeners(0);
   }
 
-  public onChange(listener: (graph: dataform.IRunResult) => Promise<void> | void): Runner {
+  public onChange(listener: (graph: dataform.IRunResult) => void): Runner {
```

Comment:
This is quite a meaningful change, I'm sure we discussed it a lot in the past, it will change behaviour quite a bit! Should be fine as we debounce (I think??) these calls ourselves just wanted to check. Does mean runs should go a bit faster in the future.

One thing that could go wrong is, the way we use a debouncer, it will only write the change if no updates happen for some period of time. That could me runs with lots of fast queries and no breaks in updates could mean we don't write new results back as frequently?

Reply:
So closed source never actually returns a `Promise`, and hasn't for a very long time, so this is actually a no-op.

Runs with fast queries shouldn't be a problem, my debouncer always writes out a value after a specified period (where the start of that period is determine by the last write time): https://source.cloud.google.com/tada-analytics/github_dataform-co_dataform-co/+/master:ts/job-service/runs.ts;l=422?q=Debouncer